### ISSUE TITLE
recognition rate is very low

### ISSUE THREAD

**Issue by lbbboy** [chpoit/utsushis-charm#37]
**Device**

  - For PC, what is your screen resolution: 1080P
  - For PC, what is your OS: win11
- Game language:Simplified Chinese  traditional Chinese 
- App version: V1.7

** Problem(s) **
Please describe the issue in as much detail.
Almost unusable, only 9 out of 50 images are recognized, and they are all wrong
![微信图片_20220710072339](https://user-images.githubusercontent.com/7271994/178125650-8c3e6fa5-7495-45ff-8ca3-894b6ab6bf0b.png)
```
{'slots': [3, 0, 0], 'skills': {}, 'errors': [['霍 層 作 化', '1', 'ParseError.MUST_FIX'], ['城 垂 跖 離 UP', '2', 'ParseError.MUST_FIX']]}
frames\frame0.png
frames\frame196.png
frames\frame208.png
frames\frame253.png
frames\frame474.png
frames\frame486.png

{'slots': [3, 1, 0], 'skills': {}, 'errors': [['不 聞', '1', 'ParseError.MUST_FIX'], ['籟 酎 怯', '2', 'ParseError.MUST_FIX']]}
frames\frame112.png
frames\frame148.png
frames\frame220.png
frames\frame522.png

{'slots': [2, 1, 0], 'skills': {}, 'errors': [['泥 霈 耐 作', '1', 'ParseError.MUST_FIX'], ['披 分 術 [ 力 ]', '1', 'ParseError.MUST_FIX']]}
frames\frame122.png
frames\frame134.png
frames\frame172.png
frames\frame183.png
frames\frame242.png
frames\frame278.png
frames\frame289.png
frames\frame312.png
frames\frame346.png
frames\frame357.png
frames\frame404.png
frames\frame439.png
frames\frame451.png
frames\frame99.png

{'slots': [1, 1, 0], 'skills': {}, 'errors': [['砲 遼 褲', '1', 'ParseError.MUST_FIX']]}
frames\frame14.png
frames\frame231.png
frames\frame266.png
frames\frame381.png
frames\frame39.png
frames\frame416.png
frames\frame499.png
frames\frame63.png
frames\frame75.png

{'slots': [1, 0, 0], 'skills': {}, 'errors': [['L E', '1', 'ParseError.MUST_FIX'], ['怎 惇', '2', 'ParseError.MUST_FIX']]}
frames\frame160.png
frames\frame28.png
frames\frame301.png
frames\frame324.png
frames\frame511.png
frames\frame547.png
frames\frame559.png

{'slots': [1, 1, 1], 'skills': {}, 'errors': [['居 佑 賭 帚 跋 慈 的 酒', '1', 'ParseError.MUST_FIX'], ['張 化 括 纏', '1', 'ParseError.MUST_FIX']]}
frames\frame335.png
frames\frame427.png
frames\frame535.png
frames\frame87.png

{'slots': [3, 2, 0], 'skills': {}, 'errors': [['耳 壑', '1', 'ParseError.MUST_FIX'], ['泡 沬 2 舞', '1', 'ParseError.MUST_FIX']]}
frames\frame369.png
frames\frame50.png

{'slots': [0, 0, 0], 'skills': {}, 'errors': [['力 星 鯨', '1', 'ParseError.MUST_FIX'], ['粟 降 寺 孵 張', '1', 'ParseError.MUST_FIX']]}
frames\frame463.png
frames\frame571.png
frames\frame584.png

```
[video](https://drive.google.com/file/d/1pl7DA0fuacfeDBW8z3BvQ4EnQxDApnKC/view?usp=sharing)

**Comment by chpoit** [comment]
As much as I want to help you, languages with kanji are very hard to parse for tesseract, espescially considering the resolution of the switch recordings. The amount of detail in a single character is quite high and as good as Google's OCR data might be, it is not perfect.

I can try to improve this, but I don't think I will be able to fully resolve the issue, you just have to look at the second character in the first frame of your video ![Screenshot_20220710_090519](https://user-images.githubusercontent.com/17057889/178146148-3a206be9-2a31-4b4f-bc07-d4e80f7fe77b.png), it's already blurry for humans, a computer will struggle even more.
![Screenshot_20220710_090519](https://user-images.githubusercontent.com/17057889/178146148-3a206be9-2a31-4b4f-bc07-d4e80f7fe77b.png)

Thank you for your video, I can at least use it to attempt to improve detection.

The "easiest" solution would be for you to add automatic corrections in the `corrections.chi_tra.csv` or `corrections.chi_sim.csv` in your `Appdata/Local/utsushis-charm` directory.

As an aside, I am curious, you seem to understand english well, why don't you record in english and use english set builders? Are you simply too used to the chinese names for everything?

**Comment by lbbboy** [comment]
I found an easier way

[MHR charm json to txt converter
](https://www.nexusmods.com/monsterhunterrise/mods/880)

**Comment by Vessile822** [comment]
src.charm_extraction - WARNING - Skill-less charm found in frames
Why cant recognize?

**Comment by chpoit** [comment]
> src.charm_extraction - WARNING - Skill-less charm found in frames Why cant recognize?

I am afraid I cannot help you with this little information. Either create a complete issue, or submit proper information related to this issue. Thank you.
